osu.Game/Rulesets/Edit/Checks/CheckDelayedHitsounds.cs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using osu.Framework.Audio.Track;
+using osu.Game.Audio;
+using osu.Game.Extensions;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckDelayedHitsounds : ICheck
+    {
+        /// <summary>
+        /// Threshold at which point the sample is considered silent.
+        /// </summary>
+        private const float silence_threshold = 0.001f;
+
+        private const float falloff_factor = 0.95f;
+        private const int delay_threshold = 5;
+        private const int delay_threshold_negligible = 1;
+
+        private readonly string[] audioExtensions = { "mp3", "ogg", "wav" };
```

Comment:
Should really be moved out to a constant at some point somewhere because it's starting to show up all over. Case in point:

https://github.com/ppy/osu/blob/d701a2596985d497096a6e85d0585456cdbfd553/osu.Game/Rulesets/Edit/Checks/CheckTooShortAudioFiles.cs#L19